pokemon-checker/src/factories/AbilityFactory.test.ts

```diff
@@ -0,0 +1,34 @@
+import { AbilityFactory } from "./AbilityFactory";
+import { pressureAbility } from "../_stubs/AbilityData";
+import { IAbilityData } from "../interfaces/PokemonData";
+import AbilityDTO from "../DataTransferObjects/AbilityDTO";
+
+const factory = new AbilityFactory();
+const pressureStub = {
+  name: pressureAbility.name,
+  url: pressureAbility.url,
+  id: -1,
+  effect: "Missing Effect",
+  pokemons: [],
+  localizedName: pressureAbility.name,
+};
+let mockStub: AbilityDTO;
+let mockFull: AbilityDTO;
+let apiRetrieved: IAbilityData;
+
+beforeAll(async () => {
+  apiRetrieved 
+    = await fetch(pressureStub.url).then(retrieved => retrieved.json());
+})
+
+test("Can create stub ability", () => {
+  mockStub = factory.createAbilityStub(pressureStub)
+  expect(mockStub).toEqual(pressureStub);
+})
+
+test("Can convert stub to full", () => {
+  mockFull = factory.createAbilityFromDataAndStub(apiRetrieved, mockStub);
```

Comment:
So one thing about unit tests is they need to be able to run in any order:
![image](https://user-images.githubusercontent.com/3711737/163768223-90f42c6d-fe10-4299-8be1-9a44e12a4a16.png)
(From the [MS docs on unit tests](https://docs.microsoft.com/en-us/visualstudio/test/unit-test-basics))


I pulled this to my machine and tried a quick check to run this test first, and things break.

Can we update the "stub to full" code to be fully independent? It could make its own stub so it doesn't need to follow the stub test.

Reply:
Right I forgot about unit tests not running in order, I must have been getting lucky in that they were